src/middleware.ts

```diff
@@ -55,12 +49,42 @@ export const createCamelResponseTransformer: CreateAxiosResponseTransformer = (
     data: unknown,
     headers?: unknown
   ): ReturnType<ReturnType<CreateAxiosResponseTransformer>> => {
-    if (!options?.ignoreHeaders) {
-      camel(headers, { overwrite: true, ...options });
-    }
+    overwriteHeaders(headers, camel, options);
     return camel(data, options);
   };
 };
+const overwriteHeaders = (
```

Comment:
I recommend `overwriteHeadersOrNoop` to mean it will do nothing in some condition